src/openrelik_mcp_server/tools.py

```diff
@@ -164,6 +165,59 @@ def read_file_content(file_id: int) -> bytes | str:
     return response.content
 
 
+def get_files_recursive(
+    root_folder_id: int,
+    filename_regex: str,
+    api_client: Any | None,
+) -> list[dict[str, Any]]:
+    """
+    Helper function to list files recursively in an OpenRelik folder filtering on a filename regular expression.
+    """
+    if not api_client:
+        api_client = get_openrelik_client()
+
+    response = api_client.get(f"/folders/{root_folder_id}/files/")
+    files = response.json()
+    all_files = []
+
+    pattern = re.compile(filename_regex)
+    for file in files:
+        if pattern.search(file["display_name"]):
+            all_files.append(file)
+
+    subfolders = api_client.get(f"/folders/{root_folder_id}/folders/").json()
+    for folder in subfolders:
+        subfolder = folder["id"]
+        files_from_subfolder = get_files_recursive(
+            subfolder, filename_regex, api_client
+        )
+        all_files.extend(files_from_subfolder)
+
+    return all_files
+
+
+@mcp.tool()
+def find_files_recursive(
+    root_folder_id: int,
+    filename_regex: str,
+) -> list[dict[str, Any]]:
+    """
+    Lists files recursively in an OpenRelik folder filtering on a filename regular expression.
+    Always returns a JSON string with the list of files with their metadata.
+
+    Args:
+        folder_id: The ID of the folder to list the files from.
```

Comment:
nit: doesn't return a JSON string but a dict. However, I'd just remove here it given it's a repeat of what's in the `Return` section of your docstring.

```suggestion
    """
    Lists files recursively in an OpenRelik folder filtering on a filename regular expression.

    Args:
        folder_id: The ID of the folder to list the files from.
```

Reply:
Done